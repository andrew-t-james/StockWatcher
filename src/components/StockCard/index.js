import React from 'react';

const StockCard = props => {
  const { name, shortName, high, low, latestPrice, change } = props;

  return (
    <section className="stock-card">
      <div className="stock-card__info">
        <div>
          <h4 className="stock-card__title">{name.length > 13 ? `${name.substring(0, 13)}...` : name}</h4>
          <p className="stock-card__short-name">{shortName}</p>
        </div>

        {change < 0 && (
          <div className="stock-card__change">
            <p className=" negative-change">
              <span className="stock-card__pointer"> &#x25bc;</span>
              {change} ({(change / latestPrice).toFixed(2)}%)
            </p>
            <p className="stock-card__price">${latestPrice}</p>
          </div>
        )}

        {change > 0 && (
          <div className="stock-card__change">
            <p className="stock-card__change positive-change">
              <span className="stock-card__pointer"> &#x25b2;</span>
              {change} ({(change / latestPrice).toFixed(2)}%)
            </p>
            <p className="stock-card__price">${latestPrice}</p>
          </div>
        )}
      </div>

      <div className="stock-card-average">
        <span className="stock-metric" style={{ top: high.toFixed(2) / 100 - change }}>
          &#x25ba;
        </span>
        <p>{high.toFixed(2)}</p>
        <p>{low.toFixed(2)}</p>
      </div>
    </section>
  );
};

export default StockCard;
