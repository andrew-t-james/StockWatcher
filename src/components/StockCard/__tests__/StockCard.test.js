import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import StockCard from '../index';

describe('StockCard Unit test', () => {
  const mockData = {
    name: 'Apple Inc.',
    shortName: 'AAPL',
    high: 188.35,
    low: 185.93,
    latestPrice: 185.24,
    change: -1.26,
  };

  let wrapper;
  beforeEach(() => (wrapper = shallow(<StockCard {...mockData} />)));
  afterEach(() => wrapper.unmount());

  test('should render conditionally render positive stock quotes', () => {
    const mockData = {
      name: 'Alphabet Inc.',
      shortName: 'GOOG',
      high: 1177.295,
      low: 1153.83,
      latestPrice: 1161.44,
      change: 8.4,
    };
    const wrapper = shallow(<StockCard {...mockData} />);
    const actual = wrapper.find('.positive-change').text();
    const expected = ' ▲8.4 (1.01%)';
    expect(actual).toBe(expected);
  });

  test('should render conditionally render negative stock quotes', () => {
    const actual = wrapper.find('.negative-change').text();
    const expected = ' ▼-1.26 (1.02%)';
    expect(actual).toBe(expected);
  });

  test('should match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
