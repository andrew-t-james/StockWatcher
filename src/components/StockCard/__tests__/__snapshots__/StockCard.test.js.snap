// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StockCard Unit test should match snapshot 1`] = `
<section
  className="stock-card"
>
  <div
    className="stock-card__info"
  >
    <div>
      <h4
        className="stock-card__title"
      >
        Apple Inc.
      </h4>
      <p
        className="stock-card__short-name"
      >
        AAPL
      </p>
    </div>
    <div
      className="stock-card__change"
    >
      <p
        className=" negative-change"
      >
        <span
          className="stock-card__pointer"
        >
           ▼
        </span>
        -1.26
         (
        -0.01
        %)
      </p>
      <p
        className="stock-card__price"
      >
        $
        185.24
      </p>
    </div>
  </div>
  <div
    className="stock-card-average"
  >
    <span
      className="stock-metric"
      style={
        Object {
          "top": 3.1435,
        }
      }
    >
      ►
    </span>
    <p>
      188.35
    </p>
    <p>
      185.93
    </p>
  </div>
</section>
`;
